### PR TITLE
#249 -  Reactive Kafka integration

### PR DESCRIPTION
**Pull request by JakubDziworski**

### CODE REVIEW

File: src/test/groovy/com/jvm_bloggers/kafka/KafkaEventListenerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        given:
+            def issueNumber = 15
+            def issue = NewsletterIssue.builder().issueNumber(issueNumber).build()
+            def event = new NewIssuePublished(issue)
+            def expectedMsgPublished = "$ISSUE_URL" + issueNumber
+        when:
+            eventListener.publishNewIssue(event)
+        then:
+            1 * messagesPublisher.publish(new NewIssuePublishedMessage(issueNumber, expectedMsgPublished), TOPIC_NAME)
+            0 * _
```
Comment (mszarlinski) [inline src/test/groovy/com/jvm_bloggers/kafka/KafkaEventListenerSpec.groovy]: It makes production code closed for further extension and should be removed.

File: src/main/java/com/jvm_bloggers/kafka/message/NewIssuePublishedMessage.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.jvm_bloggers.kafka.message;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import lombok.Value;
+
+@KafkaMessage
+@Value
+public class NewIssuePublishedMessage {
+    @JsonProperty("issueNumber")
```
Comment (goostleek) [inline src/main/java/com/jvm_bloggers/kafka/message/NewIssuePublishedMessage.java]: Are these `@JsonProperty` annotations really required if the field names are equal to the annotation value?

Comment (JakubDziworski) [inline src/main/java/com/jvm_bloggers/kafka/message/NewIssuePublishedMessage.java]: They are meant to protect against refactoring (if someone changes field name).

However tests would also detect this too, so I agree - they are redundant.

File: src/main/java/com/jvm_bloggers/kafka/serialization/MessageSerializer.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.jvm_bloggers.kafka.exception.UnableToSerializeMessageException;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class MessageSerializer {
+    private final ObjectMapper objectMapper;
+
+    public MessageSerializer() {
+        objectMapper = new ObjectMapper();
```
Comment (JakubDziworski) [inline src/main/java/com/jvm_bloggers/kafka/serialization/MessageSerializer.java]: The way Java's akka streams makes it very unreadable. For example the type of testProbe is actually  ``` TestSubscriber.Probe<ConsumerMessage.CommittableMessage<byte[], String>>``` :D. Before comminting I spoke to Tomek about this issue and we agreed that we make an exception and leave it with def.

This is how it'd look with types:

```java
ConsumerSettings<byte[], String> consumerSettings = ConsumerSettings
                    .create(actorSystem, new ByteArrayDeserializer(), new StringDeserializer())
                    .withBootstrapServers("127.0.0.1:9092")
                    .withGroupId("kafkaTestGroup")
                    .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
            NewIssuePublishedMessage publishedMessage = new NewIssuePublishedMessage(15L, "http://jvm-bloggers.com/issue/15");
            TestSubscriber.Probe<ConsumerMessage.CommittableMessage<byte[], String>> testProbe = Consumer.
                    committableSource(consumerSettings, Subscriptions.topics("com.jvm_bloggers.issue.published"))
                    .toMat(TestSink.probe(actorSystem), Keep.right())
                    .run(actorMaterializer);
```

**Comment by JakubDziworski** [comment]
Thanks for reviewing! All requested changes has been applied.

**Pull request merged**